### DISCUSSION TITLE
ARQ

### DISCUSSION THREAD

**Discussion by lvplummer** [jgromes/RadioLib#749] [Ideas]
Have you considered adding a general purpose ARQ layer that could be used with multiple hardware types?

Larry

**Comment by jgromes** [comment]
In generla, I try to stay away from implementing these things directly in RadioLib. The logic is that the original purpose of the library was to be a driver for the different transceiver ICs - the library would not do anything that the module itself could not.

The obvious and significant objection is the entire `protocols` directory. This mainly has ham radio protocols/digital modes, since these usually require significant "hacks" to work with off-the-shelf transceiver ICs. So, if there is a standardized ARQ protocol that could be  implemented, the it could be added if there is significant enough user base. Otherwise, I think it's a bit out-of-scope for the lbirasry to create its own protocol.
